wrappers/binaries.go

```diff
@@ -21,13 +21,71 @@
 package wrappers
 
 import (
+	"bufio"
 	"os"
+	"regexp"
+	"strconv"
 
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/osutil"
 	"github.com/snapcore/snapd/snap"
 )
 
+// detectLegacy checks whether bash-completion is version 2.2 or later.
+// Prior to 2.2, bash-completion does not support XDG_DATA_DIRS.
+// If it fails to detect the version of bash-completion, it disables
+// completion.
```

Comment:
It might be easier if this function returned an enum,

    NoCompletion,
    LegacyCompletion,
    FullCompletion,

Reply:
Done